nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/controller/state/server/TestZooKeeperStateServerConfigurations.java

```diff
@@ -0,0 +1,193 @@
+        final NiFiProperties insecureProps = NiFiProperties.createBasicNiFiProperties(null, INSECURE_PROPS);
+        final ZooKeeperStateServer server = ZooKeeperStateServer.create(insecureProps);
+
+        Assert.assertNotNull(server);
+        Assert.assertNotNull(server.getQuorumPeerConfig().getClientPortAddress());
+    }
+
+    // This test shows that the client can specify a secure port and that port is used:
+    @Test
+    public void testCreateWithSpecifiedSecureClientPort() throws IOException, QuorumPeerConfig.ConfigException {
+        final NiFiProperties secureProps = NiFiProperties.createBasicNiFiProperties(null, new HashMap<String, String>() {{
+            putAll(SECURE_NIFI_PROPS);
+            put(ZOOKEEPER_PROPERTIES_FILE_KEY, SECURE_ZOOKEEPER_PROPS);
+        }});
+
+        final ZooKeeperStateServer server = ZooKeeperStateServer.create(secureProps);
+        Assert.assertNotNull(server);
+
+        final QuorumPeerConfig config = server.getQuorumPeerConfig();
+        Assert.assertEquals(secureZooKeeperProps.getProperty("secureClientPort"), String.valueOf(config.getSecureClientPortAddress().getPort()));
+    }
+
+    // This shows that a secure NiFi with an secure ZooKeeper will not have an insecure client address or port:
+    @Test
+    public void testCreateRemovesInsecureClientPort() {
+        Assert.assertNotNull(secureZooKeeperProps.getProperty("secureClientPort"));
+        Assert.assertNotEquals(secureZooKeeperProps.getProperty("clientPort"), "");
+        Assert.assertNull(secureQuorumPeerConfig.getClientPortAddress());
+    }
+
+    // This test shows that a connection class is set when none is specified (QuorumPeerConfig::parseProperties sets the System property):
+    @Test
+    public void testCreateWithUnspecifiedConnectionClass() {
+        Assert.assertEquals(org.apache.zookeeper.server.NettyServerCnxnFactory.class.getName(), System.getProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY));
+    }
+
+    // This test shows that a specified connection class is honored (QuorumPeerConfig::parseProperties sets the System property):
+    @Test
+    public void testCreateWithSpecifiedConnectionClass() throws IOException, QuorumPeerConfig.ConfigException {
+        final NiFiProperties secureProps = NiFiProperties.createBasicNiFiProperties(null, new HashMap<String, String>() {{
+            putAll(SECURE_NIFI_PROPS);
+            put(ZOOKEEPER_PROPERTIES_FILE_KEY, SECURE_ZOOKEEPER_PROPS);
+        }});
+
+        Assert.assertNotNull(ZooKeeperStateServer.create(secureProps));
+        Assert.assertEquals(ZOOKEEPER_CNXN_FACTORY, System.getProperty(ServerCnxnFactory.ZOOKEEPER_SERVER_CNXN_FACTORY));
+    }
+
+    // ZooKeeper client and quorum TLS enabled by hand in zookeeper.properties
+
+    private CloseableHttpClient httpclient;
+
+    @Before
+    public void setUp() throws Exception {
+        httpclient = HttpClients.createDefault();
```

Comment:
Should this declaration be moved to the `setupWithValidProperties()` method to avoid multiple `@Before` annotated methods?